src/Diagnostics.RuntimeHost/Middleware/DiagnosticsRequestMiddleware.cs

```diff
@@ -83,6 +83,7 @@ private void BeginRequest_Handle(HttpContext httpContext)
             var dataSourcesConfigurationService = ((ServiceProvider)httpContext.RequestServices).GetService<IDataSourcesConfigurationService>();
             var wawsObserverTokenService = ((ServiceProvider)httpContext.RequestServices).GetService<IWawsObserverTokenService>();
             var supportBayApiObserverTokenService = ((ServiceProvider)httpContext.RequestServices).GetService<ISupportBayApiObserverTokenService>();
+            var kustoHeartBeatService = ((ServiceProvider)httpContext.RequestServices).GetService<IKustoHeartBeatService>();
```

Comment:
This is actually an AntiPattern called ServiceLocator that I started with the wawsObserverTokenService. Instead of using the `GetService` method to get the service we want we should take advantage of the dependency injecting capabilities of .NET Core. We can proceed with this behavior for the sake and urgency of this PR but lets correct this behavior in a separate PR. I'll create a work item for this in our backlog.

Reply:
I have a [User Story](https://dev.azure.com/app-service-diagnostics-portal/app-service-diagnostics-portal/_workitems/edit/341/) but haven't added this file to it yet, I'll update it